src/Microsoft.ML.Core/Data/ColumnType.cs

```diff
@@ -940,6 +957,19 @@ public override bool Equals(ColumnType other)
             return true;
         }
 
+        public override bool Equals(Object other)
+        {
+            var tmp = other as ColumnType;
+            if (tmp == null)
+                return false;
+            return Equals(tmp);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hashing.CombinedHash(_itemType.GetHashCode(), _size, Utils.Size(_dims), _dims);
```

Comment:
>_dims [](start = 91, length = 5)

You can't do that... two distinct arrays with equal items do *not* equal each other. Similarly, the hash code of an array is not some sort of combined hash of its elements, but is rather based on the general kinda-pointer-based hash that is the default for objects. Rather you will have to have some scheme for combining the hash of each individual element.

Also: once you do that there is no point in hashing `_size` and the length of `_dims` since those are ultimately just derived from the array anyway.

Also also: While it's moot since I don't think you should have done it, please don't use `Utils.Size` on an array unless it might *conceivably* be `null`. (It won't be in this case.)